#!/bin/bash

# This is a script to assemble the transcripts from a SAM/BAM file using String tie

# First load the conda module

#conda activate assembly

# Move into the appropriate directory

#cd ~/lncRNA_2024/02_assembly/stringtie


# Start the assembly

# Stringtie recommends following this workflow:

# Run stringtie normally, with the -G option (reference annotation to include in the merging (GTF/GFF3))

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/C_6_II_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/C_6_II.gtf

#echo "Finished C_6_II.gtf"


#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/C_24_II_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/C_24_II.gtf

#echo "Finished C_24_II.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/C_6_III_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/C_6_III.gtf

#echo "Finished C_6_III.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/C_24_III_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/C_24_III.gtf

#echo "Finished C_24_III.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/T_6_II_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/T_6_II.gtf

#echo "Finished T_6_II.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/T_24_II_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/T_24_II.gtf

#echo "Finished T_24_II.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/T_6_III_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/T_6_III.gtf

#echo "Finished T_6_III.gtf"

#stringtie /media/ana/6E24AF8524AF4EC1/sam_files/T_24_III_Aligned.sortedByCoord.out.bam -G /home/ana/lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 \
#-o ~/lncRNA_2024/02_assembly/stringtie/temp/T_24_III.gtf

#echo "Finished T_24_III.gtf"

# Run StringTie with --merge in order to generate a non-redundant set of transcripts observed in any of the RNA-Seq samples assembled previously. 
# The stringtie --merge mode takes as input a list of all the assembled transcripts files (in GTF format) previously obtained for each sample, 
# as well as a reference annotation file (-G option) if available.

#stringtie --merge temp/C_6_II.gtf temp/C_24_II.gtf temp/C_6_III.gtf temp/C_24_III.gtf temp/T_6_II.gtf temp/T_24_II.gtf temp/T_6_III.gtf temp/T_24_III.gtf -G /home/ana/#lncRNA_2024/reference_genome/Zea_mays_B73_RefGen_v5.exon.gtf -p 12 -o stringtie/Merged_samples.gtf

#echo "Finished merging"


# For each sample, run StringTie using the -B/-b and -e options in order to estimate transcript abundances and generate read coverage tables for Ballgown. 
# The -e option is not required but recommended for this run in order to produce more accurate abundance estimations of the input transcripts. 

#Each StringTie run in this step will take as input the SAM/BAM file for the corresponding sample and the -G option with the merged transcripts (GTF file) generated by stringtie --merge in the previous step. 


stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/C_6_II_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Lt_C_6.gtf

echo "Finished Lt_C_6.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/C_24_II_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Lt_C_24.gtf

echo "Finished Lt_C_24.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/C_6_III_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Ls_C_6.gtf

echo "Finished Ls_C_6.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/C_24_III_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Ls_C_24.gtf

echo "Finished Ls_C_24.gtf"



stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/T_6_II_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Lt_T_6.gtf

echo "Finished Lt_T_6.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/T_24_II_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Lt_T_24.gtf

echo "Finished Lt_T_24.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/T_6_III_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Ls_T_6.gtf

echo "Finished Ls_T_6.gtf"

stringtie -Be /media/ana/6E24AF8524AF4EC1/sam_files/T_24_III_Aligned.sortedByCoord.out.bam -G Merged_samples.gtf -p 12 \
-o ~/lncRNA_2024/02_assembly/stringtie/Ls_T_24.gtf

echo "Finished Ls_T_24.gtf"
